### PR TITLE
Ignoring Android scanner errors resulting from icon lookup failures.

### PR DESCRIPTION
**Pull request by lpusok**
- If icon format is unsupported, does not fail, instead returns default icons.
- Send failures to analytics Info level log, does not return warnings to user

### CODE REVIEW

File: analytics/analytics_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+			},
+		},
+		{
+			name: "nil data",
+			data: nil,
+			want: map[string]interface{}{
+				"source": "scanner",
+			},
+		},
+		{
```
Comment (godrei) [inline analytics/analytics_test.go:27]: NOTE: might a test case which represents that we set (overwrite the existing value of) `source` would be useful.

File: scanners/android/icon.go [hunk 2]
```diff
@@ -25,30 +26,23 @@ func lookupIconName(manifestPth string) ([]icon, error) {
...
-		log.Debugf("Key application not found in manifest file")
-		return nil, nil
+		return nil, fmt.Errorf("Key application not found in manifest file")
 	}
 	ic := app.SelectAttr("android:icon")
 	if ic == nil {
-		log.Debugf("Attribute not found in manifest file")
-		return nil, nil
+		// Gradle varaibles like ${appIcon} are not supported
+		return nil, fmt.Errorf("Attribute not found in manifest file")
```
Comment (godrei) [inline scanners/android/icon.go:45]: error string should not be capitalized: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Comment (godrei) [reply scanners/android/icon.go:45]: NOTE: I think mentioning which attribute (`android:icon`) is missing would make this error more informative

File: scanners/android/icon.go [hunk 3]
```diff
@@ -105,7 +99,7 @@ func lookupIcons(projectDir string, basepath string) ([]string, error) {
 	for _, manifestPath := range manifestPaths {
 		icons, err := lookupIconName(manifestPath)
 		if err != nil {
-			return nil, err
+			analytics.LogInfo("android-icon-lookup", analytics.DetectorErrorData("android", err), "Failed to lookup android icon from manifest")
```
Comment (godrei) [inline scanners/android/icon.go:102]: missing continue

**Review by godrei — CHANGES_REQUESTED** [review]
Pls see my notes

File: scanners/android/icon.go (hunk 3 above)
Comment (lpusok) [reply scanners/android/icon.go:102]: append seems to be working without an explicit continue, but maybe the intent is clearer that way?

**Review by Szabadember — APPROVED** [review]

File: scanners/android/icon.go (hunk 3 above)
Comment (godrei) [reply scanners/android/icon.go]: i think so

**Review by godrei — APPROVED** [review]

**Pull request merged**